gremlin-dotnet/src/Gremlin.Net/Process/Traversal/ITraversalSideEffects.cs

```diff
@@ -29,6 +29,7 @@ namespace Gremlin.Net.Process.Traversal
     /// <summary>
     ///     A <see cref="ITraversal" /> can maintain global sideEffects.
     /// </summary>
+    [Obsolete("As of release 3.3.8, not replaced, prefer use of cap()-step to retrieve side-effects as part of traversal iteration")]
```

Comment:
Is the intention to deprecate `ITraversalSideEffects`?

Reply:
hmm - i think so. it's going to end up being a do-nothing interface ultimately. If we removed `ITraversal.SideEffects` then it wouldn't serve much purpose. We don't do any local side-effects, so without remote ones it doesn't seem useful. Unless we just keep it around for API completeness to match Java?

Reply:
ahh, I see what you mean, makes sense to deprecate the entire interface from the GLV.